### ISSUE TITLE
How to run tests and how to generate documentation

### ISSUE THREAD

**Issue by huseyinyilmaz** [tuncer/ememo#1]
Hello,
First of all this is a very good project to learn things like coding conventions, testing, documentation, dialyzer usage etc.

I would normally look at the make file but you do not have one.
Can you tell me:
* How to run tests
* How to generate documentation.
* How to run dialyzer. (Here is how I did it. I just want to learn if you are doing something differently.)
```
Huseyins-Air:ememo huseyin$ dialyzer --build_plt --apps erts kernel stdlib crypto mnesia sasl eunit
  Compiling some key modules to native code... done in 0m37.43s
  Creating PLT /Users/huseyin/.dialyzer_plt ...
eunit_test.erl:305: Call to missing or unexported function eunit_test:nonexisting_function/0
Unknown functions:
  compile:file/2
  compile:forms/2
  compile:noenv_forms/2
  compile:output_generated/1
  xref:add_application/3
  xref:analyze/2
  xref:set_default/3
  xref:set_library_path/2
  xref:start/2
  xref:stop/1
Unknown types:
  compile:option/0
 done in 1m21.03s
done (warnings were emitted)
Huseyins-Air:ememo huseyin$ dialyzer src/*
  Checking whether the PLT /Users/huseyin/.dialyzer_plt is up-to-date... yes
  Proceeding with analysis... done in 0m0.66s
done (passed successfully)
```

Thanks.

**Comment by tuncer** [comment]
Thanks for noticing, I've updated README and CONTRIBUTING.

**Comment by tuncer** [comment]
By using abbreviated commands, you can also (1) compile, (2) test, and (3) dialyze the code like this:

```
$ rebar b-p
$ rebar co ct di
```

This is shorter to type. See `rebar help` for a description of abbreviated command support.

**Comment by huseyinyilmaz** [comment]
@tuncer Are those rebar 3 commands. I am getting following errors with rebar 2
```
Huseyins-Air:ememo huseyin$ ./rebar --version
rebar 2.5.1 R14B04 20140730_173812 git 2.5.1-1-ge9f62c4
Huseyins-Air:ememo huseyin$ ./rebar b-p
WARN:  ''b-p'' command does not apply to directory /Users/huseyin/projects/ememo
Command ''b-p'' not understood or not applicable
Huseyins-Air:ememo huseyin$ ./rebar co ct di
==> ememo (compile)
==> ememo (ct)
DONE.
Testing projects.ememo: TEST COMPLETE, 3 ok, 0 failed of 3 test cases

WARN:  'di' command does not apply to directory /Users/huseyin/projects/ememo
Command 'di' not understood or not applicable
Huseyins-Air:ememo huseyin$
```

**Comment by tuncer** [comment]
You need current master or the soon to be released 2.6.0.
